yt_dlp/extractor/youtube.py

```diff
@@ -627,32 +632,66 @@ def _extract_context(self, ytcfg=None, default_client='web'):
         client_context.update({'hl': self._preferred_lang or 'en', 'timeZone': 'UTC', 'utcOffsetMinutes': 0})
         return context
 
-    _SAPISID = None
+    @staticmethod
+    def _make_sid_authorization(scheme, sid, origin, additional_parts):
+        timestamp = str(round(time.time()))
+
+        hash_parts = []
+        if additional_parts:
+            hash_parts.append(':'.join(additional_parts.values()))
+        hash_parts.extend([timestamp, sid, origin])
+        sidhash = hashlib.sha1(' '.join(hash_parts).encode()).hexdigest()
+
+        parts = [timestamp, sidhash]
+        if additional_parts:
+            parts.append(''.join(additional_parts))
+
+        return f'{scheme} {"_".join(parts)}'
 
-    def _generate_sapisidhash_header(self, origin='https://www.youtube.com'):
-        time_now = round(time.time())
-        if self._SAPISID is None:
-            yt_cookies = self._get_cookies('https://www.youtube.com')
+    _SAPISID = _3PSAPISID = _1PSAPISID = None
+
+    def _load_sid_cookies(self):
+        yt_cookies = self._get_cookies('https://www.youtube.com')
+        yt_sapisid = try_call(lambda: yt_cookies['SAPISID'].value)
+        yt_3papisid = try_call(lambda: yt_cookies['__Secure-3PAPISID'].value)
+
+        if not self._SAPISID:
             # Sometimes SAPISID cookie isn't present but __Secure-3PAPISID is.
+            # YouTube also falls back to __Secure-3PAPISID if SAPISID is missing.
             # See: https://github.com/yt-dlp/yt-dlp/issues/393
-            sapisid_cookie = dict_get(
-                yt_cookies, ('__Secure-3PAPISID', 'SAPISID'))
-            if sapisid_cookie and sapisid_cookie.value:
-                self._SAPISID = sapisid_cookie.value
-                self.write_debug('Extracted SAPISID cookie')
-                # SAPISID cookie is required if not already present
-                if not yt_cookies.get('SAPISID'):
-                    self.write_debug('Copying __Secure-3PAPISID cookie to SAPISID cookie')
-                    self._set_cookie(
-                        '.youtube.com', 'SAPISID', self._SAPISID, secure=True, expire_time=time_now + 3600)
-            else:
-                self._SAPISID = False
-        if not self._SAPISID:
+            self._SAPISID = yt_sapisid or yt_3papisid
+
+        if not self._1PSAPISID:
+            self._1PSAPISID = try_call(lambda: yt_cookies['__Secure-1PAPISID'].value)
+
+        if not self._3PSAPISID:
+            self._3PSAPISID = yt_3papisid
```

Comment:
This will cache invalidated cookie values and send them with the API calls. It will make it difficult for us to debug what is happening when cookies are rotated, and it will hammer the API with invalid auth headers which could be detrimental to users' account standing

Would be better to not cache at all IMO

Reply:
Removed caching logic